src/libostree/ostree-repo-pull.c

```diff
@@ -5013,7 +5013,8 @@ ostree_repo_find_remotes_async (OstreeRepo                     *self,
         default_finders[finder_index++] = finder_mount = OSTREE_REPO_FINDER (ostree_repo_finder_mount_new (NULL));
 
 #ifdef HAVE_AVAHI
-      if (self->repo_finders == NULL || g_strv_contains ((const char * const *)self->repo_finders, "lan"))
+      /* Only check the LAN if it was explicitly enabled (it's more expensive) */
+      if (self->repo_finders != NULL && g_strv_contains ((const char * const *)self->repo_finders, "lan"))
```

Comment:
Would it make more sense to have `self->repo_finders` default to `config` and `mount` instead of treating `lan` specially here? Because this is inconsistent with what `ostree_repo_get_repo_finders()` would say.

Reply:
Well `ostree_repo_get_repo_finders()` would return `NULL` which means "the default set which is up to libostree". If we were to set `repo->repo_finders = "config,mount"` as a default, the user couldn't do `ostree find-remotes --finders=lan ...` (and similar for the API that lets you pass finders in)

Reply:
> Well `ostree_repo_get_repo_finders()` would return `NULL` which means "the default set which is up to libostree".

Right, but that default policy is now inconsistent with this change.

> If we were to set `repo->repo_finders = "config,mount"` as a default, the user couldn't do `ostree find-remotes --finders=lan ...` (and similar for the API that lets you pass finders in)

I think it might be better to change the policy that the config key is implementing. Right now the key is both the default set of finders _and_ the bounding set of finders. If it was just the default set (with `config` and `mount`), then `ostree find-remotes --finders=lan` would be allowed, it just wouldn't be the default. I think this makes the policy handled by that option clearer.

If you want to have an option that controls the bounding set (for someone that wants to actively disallow a finder), then I think that should be a different option. `core.allowed-repo-finders` or something.

Reply:
Sure that makes sense. I think having `core.repo-finders` be the default set not the bounding set will serve the right purpose. Then Flatpak can use the value of `ostree_repo_get_repo_finders` in the implementation of `flatpak_installation_list_remotes_by_type`, which is the only place Flatpak calls `ostree_repo_find_remotes_async` with a non-NULL set of `finders`.